### PR TITLE
Add state_color to UI editors

### PR DESCRIPTION
**Pull request by iantrich**
## Breaking change

Add state_color option to UI editors; glance/entities/button

## Type of change

- [ ] Dependency upgrade
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue: 
- Link to documentation pull request: 

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Comment by iantrich** [comment]
For next beta cut

File: src/panels/lovelace/cards/hui-entities-card.ts [hunk 1]
```diff
@@ -36,7 +36,7 @@ class HuiEntitiesCard extends LitElement implements LovelaceCard {
   }
 
   public static getStubConfig(): object {
-    return { entities: [] };
+    return { entities: [], state_color: false };
```
Comment (bramkragten) [inline src/panels/lovelace/cards/hui-entities-card.ts]: Don't add it to the stub, it is not a required setting

File: src/panels/lovelace/editor/config-elements/hui-button-card-editor.ts [hunk 2]
```diff
@@ -69,6 +71,10 @@ export class HuiButtonCardEditor extends LitElement
     return this._config!.show_icon || true;
   }
 
+  get _state_color(): boolean {
```
Comment (bramkragten) [inline src/panels/lovelace/editor/config-elements/hui-button-card-editor.ts]: You don't use this? I think you can remove it?

File: src/panels/lovelace/editor/config-elements/hui-entities-card-editor.ts [hunk 3]
```diff
@@ -80,29 +85,37 @@ export class HuiEntitiesCardEditor extends LitElement
...
-          @change="${this._valueChanged}"
+          .checked=${this._config!.show_header_toggle !== false}
+          .configValue=${"show_header_toggle"}
+          @change=${this._valueChanged}
           >${this.hass.localize(
             "ui.panel.lovelace.editor.card.entities.show_header_toggle"
           )}</ha-switch
         >
+        <ha-switch
+          .checked=${this._config!.state_color !== false}
```
Comment (bramkragten) [inline src/panels/lovelace/editor/config-elements/hui-entities-card-editor.ts]:
```suggestion
          .checked=${this._config!.state_color === true}
```

File: src/translations/en.json [hunk 4]
```diff
@@ -1872,6 +1872,7 @@
               "show_icon": "Show Icon?",
               "show_name": "Show Name?",
               "show_state": "Show State?",
+              "state_color": "Show State Color?",
```
Comment (bramkragten) [inline src/translations/en.json]:
```suggestion
              "state_color": "Color icons based on state?",
```

File: src/panels/lovelace/cards/hui-entities-card.ts (hunk 1 above)
Comment (iantrich) [reply src/panels/lovelace/cards/hui-entities-card.ts]: If you don't then the editor shows it as true on the creation of a new one

File: src/panels/lovelace/editor/config-elements/hui-entities-card-editor.ts (hunk 3 above)
Comment (iantrich) [reply src/panels/lovelace/editor/config-elements/hui-entities-card-editor.ts]: Oh maybe that was my problem

**Pull request closed without merging**